src/routes/email/guards/email-deletion.guard.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  Injectable,
+} from '@nestjs/common';
+import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+import { verifyMessage } from 'viem';
+
+/**
+ * The EmailDeletionGuard guard should be used on routes that require
+ * authenticated actions for deleting email addresses.
+ *
+ * This guard therefore validates if the message came from the specified account.
+ *
+ * The following message should be signed:
+ * email-register-${chainId}-${safe}-${emailAddress}-${account}-${timestamp}
```

Comment:
`email-delete-`

Reply:
Corrected in 27eaae4e907918ecba4248fe5b52aceb7c32a10e.